apps/backend/src/auth/OAuthAuthorization.ts

```diff
@@ -102,17 +103,46 @@ export abstract class OAuthAuthorization extends UserAuthorization {
   }
 
   private async getUserInstitutionId(userInfo: UserinfoResponse) {
-    if (userInfo.institution_name) {
-      const institutions = await this.adminDataSource.getInstitutions({
-        name: userInfo.institution_name as string,
-      });
+    const isOAuthLogin =
+      userInfo.institution_ror_id ||
+      userInfo.institution_name ||
+      userInfo.institution_country;
+
+    if (!isOAuthLogin) {
+      return undefined;
+    }
 
-      if (institutions.length === 1) {
-        return institutions[0].id;
+    let institution = userInfo.institution_ror_id
+      ? await this.adminDataSource.getInstitutionByRorId(
+          userInfo.institution_ror_id as string
+        )
+      : await this.adminDataSource.getInstitutionByName(
+          userInfo.institution_name as string
+        );
+
+    if (!institution) {
+      let institutionCountry = userInfo.institution_country
+        ? await this.adminDataSource.getCountryByName(
+            userInfo.institution_country as string
+          )
+        : null;
+
+      if (!institutionCountry || !institutionCountry.countryId) {

```

Comment:
Do we need to check both institutionCountry or institutionCountry.countryId? All countries does have an id?

Reply:
We don't need to check institutionCountry.countryId here. 
InstitutionCountry value can only be `Country | null` type. I will remove the redundant condition